src/release/breaking-changes/boxdecoration-shape-migration.md

```diff
@@ -0,0 +1,108 @@
+---
+title: Migration guide for BoxDecoration.circle
+description: Removal of BoxDecoration.circle and how to migrate
+---
+
+## Summary
+
+This change removes `BoxDecoration(shape: BoxShape.circle)`. Existing apps
+should use `ShapeDecoration(shape: CircleBorder())` instead.
+
+## Context
+
+`BoxDecoration(shape: BoxShape.circle)` uses an enum. Each new enum value requires new code
+in many places. That new code may not shake out,
+since the compiler doesn't know which values never get used.
+`BoxDecoration` has a permanent binary size cost,
+even for people who don't use it.
+Flutter moved to `ShapeBorders` rather than `BoxShape` for new
+shapes, and the existence of `BoxShape` remains as a legacy feature.
+
+Use `ShapeDecoration`. It aligns with the rest of the Flutter API.
+`BoxShape` has been deprecated.
+Continuing to use that feature can cause confusion.
```

Comment:
This reads a bit odd.

I rearranged it a bit, feel free to use or ignore..

```suggestion
The enum `BoxShape` has been deprecated, as increasing the number of supported shapes through enums created unnecessary complexity and performance costs. Users should migrate to use `ShapeDecoration` in conjunction with `ShapeBorder`, the replacement for `BoxShape`.
```

> `BoxShape` has been deprecated.

This isn't in the proposed change.

> `BoxShape` remains as a legacy feature.

This isn't necessarily true. If it is deprecated, we will remove it later.

Reply:
Technically the property `shape` in BoxDecoration has been deprecated, the enum is still there for InkWell which has nothing to do with it, but they use the same enum.

Reply:
I tried to do:

> The enum property `BoxShape shape` from `BoxDecoration` has been deprecated,
as increasing the number of supported shapes through enums created
unnecessary complexity and performance costs.
Users should migrate to use `ShapeDecoration` in conjunction with
`ShapeBorder`, the replacement for `BoxShape`.